packages/components/src/components/form/Select/index.tsx

```diff
@@ -161,10 +182,134 @@ const Select = ({
+            focusedValue: null,
+            focusedOption: option,
+        });
+    };
+
+    const onKeyDown = async (e: React.KeyboardEvent) => {
+        // This function is executed when user presses keyboard
+        if (useKeyPressScroll) {
+            // Get char value of pressed key
+            const charValue = String.fromCharCode(e.keyCode);
+
+            // Get current timestamp and check how long it is since the last keyPress event happened.
+            const currentTimestamp = new Date().getTime();
+            const timeSincePreviousKeyPress = currentTimestamp - lastKeyPressTimestamp.current;
+
+            // Save current timestamp to lastKeyPressTime variable
+            lastKeyPressTimestamp.current = currentTimestamp;
+
+            // If user didn't type anything for 1.7 seconds, set searchedValue to just pressed button, otherwise add the new value to the old one
+            if (timeSincePreviousKeyPress > 1700) {
+                searchedTerm.current = charValue;
+            } else {
+                searchedTerm.current += charValue;
+            }
+
+            if (selectRef) {
+                // Get options object
+                const { options } = selectRef.current.select.props;
+
+                if (options && options.length > 1) {
+                    /* 
+                    First, check if the options are divided into sub-categories. 
+                    For example <NetworkSelect> has options divided into sub-categories "mainnet" and "testnet".  
+                    In such scenario I need to loop through all of the sub-categories and try to find appropriate option in them as well.
+                    */
+
+                    // array of all options in which I want to find the searched term
+                    let optionsToSearchThrough: Array<Option> = [];
+
+                    if (options[0].options) {
+                        // If the if() statement is true, the options are nested.
+                        // Loop through all of the sub-categories and concatenate them into one array
+                        for (let i = 0; i < options.length; i++) {
+                            optionsToSearchThrough = optionsToSearchThrough.concat(
+                                options[i].options
+                            );
+                        }
+                    } else {
+                        // If the options aren't divided into sub-categories, you can use the default options array that is present on "selectRef"
+                        optionsToSearchThrough = options;
+                    }
+
+                    // Find the option
+                    const optionToFocusOn = findOption(
+                        optionsToSearchThrough,
+                        searchedTerm.current
+                    );
+
+                    // If no option to focus on was found, clear the searchedValue
+                    if (!optionToFocusOn) searchedTerm.current = '';
```

Comment:
hmm shouldn't we reset search string only after the period of inactivity regardless of whether there is a match?

Reply:
Was not sure about this, because then the user can type "_dsvnsivndsopjvioosvv_" forever and the searchedTerm gets never cleared.  But yeah, the reality is that the user will never type "_dsvnsivndsopjvioosvv_", so I don't know.